### ISSUE TITLE
[FEA] Exception message when opening timezone file should contain timezone name

### ISSUE THREAD

**Issue by jlowe** [rapidsai/cudf#8795]
**Is your feature request related to a problem? Please describe.**
While trying to load an ORC file with libcudf, an exception was thrown that looks like this:
```
21/07/16 09:37:30 WARN TaskSetManager: Lost task 0.0 in stage 0.0 (TID 0) (172.25.0.3 executor 3): ai.rapids.cudf.CudfException: cuDF failure at: /home/jenkins/agent/workspace/jenkins-cudf-release-17-cuda11/cpp/src/io/orc/timezone.cpp:136: Failed to open the timezone file.
        at ai.rapids.cudf.Table.readORC(Native Method)
        at ai.rapids.cudf.Table.readORC(Table.java:829)
        at com.nvidia.spark.rapids.GpuOrcPartitionReader.$anonfun$readToTable$4(GpuOrcScan.scala:853)
[...]
```

Unfortunately it's not easy to determine _which_ timezone file was causing the error.

**Describe the solution you'd like**
The exception message thrown when accessing a timezone file should contain the path to the timezone file.

**Describe alternatives you've considered**
Instead of reporting the full pathname it could report just the timezone name, but having the full path may better help users determine which system package they should install in order to properly process the timezone that appears in the ORC file.

**Comment by jlowe** [comment]
Still desired for this to be fixed.

**Comment by jlowe** [comment]
Still desired.

**Comment by bdice** [comment]
This will be resolved in #17102.
